core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -901,7 +901,7 @@ class DAGScheduler(
       // the stage as completed here in case there are no tasks to run
       markStageAsFinished(stage, None)
 
-      val debugString = stage match {
+      def debugString: String = stage match {
```

Comment:
Is this really expensive to compute? it doesn't look that heavy. But you can already just pass the expression as the argument to `logDebug` and I think it will be lazily evaluated? Only mostly sure about that.